_posts/2017-07-28-reproducible-research-badges.md

```diff
@@ -0,0 +1,154 @@
+In the extender, a critical feature is the detection of the DOI just from a research paper title. For some websites such as DOAJ.org or ScienceOpen.com this is not necessary, as they provide the DOI directly for each entry. But when the DOI is not directly provided, the extender tries to get the DOI from a request to CrossRef.org. This is not alway successful or may find incorrect results depending on the paper title.
+As discussed above, in an ideal world the o2r-extender Chrome extension would not be necessary. A “hacky” solutions is not always a bad solution, but there are a few tricky parts with a workaround like this: The extension is supporting nine different websites. If there are changes to one of these, the extender has to be updated as well. For example, [Sciebo](http://sciebo.de/), an [OwnCloud](https://owncloud.org/) implementation, recently changed their URLs to include a “fileid” parameter which resulted in an error when parsing the current folder path.
+
+## Future Work
+
+There is still potential for future improvements: one of the biggest current issues is the reliability on external services such as Crossref and DOAJ. While this issue cannot be directly resolved, it can be mitigated by having the option to request multiple backend services, which could provide the same information per badge type. Not all research services are available all the time, and not relying on just a single external service improves fault tolerance. Caching might be another way to reduce this effect.
+Furthermore, the reliability on the o2r platform itself is another issue: *Licence*, *executable*, and *spatial* badges are dependent on an existing ERC, which must be linked via DOI to a publication. If a research paper has not been made available as an ERC in the o2r platform, these badge types will return “n/a” badges indicating no information. The implementation of multiple services per badge type would help with at least one of these (the licence badges) as DOAJ also offers licence information for research publications.
+
+The o2r-extender is currently only available for Google Chrome / Chromium. But since Firefox is switching to [WebExtensions](https://developer.mozilla.org/en-US/Add-ons/WebExtensions) and slowly moving away from their old “Addons” completely with [Firefox 57](https://developer.mozilla.org/en-US/Add-ons/Overlay_Extensions/Firefox_addons_developer_guide), a port from a Chrome Extension to the open WebExtensions may help making the extender available for more users. There are only minor differences between the two types of extensions, which means the port should be possible with a few changes.
+
+Other issues include:
```

Comment:
"steps for further improvement", instead of "issues".
Please also add a link to GitHub issues, ask for feedback or ideas.